sklearn/manifold/tests/test_t_sne.py

```diff
@@ -256,12 +263,13 @@ def grad(params):
     assert_almost_equal(check_grad(fun, grad, X_embedded.ravel()), 0.0, decimal=5)
 
 
-def test_trustworthiness():
+@pytest.mark.parametrize("dtype", DTYPES)
+def test_trustworthiness(dtype):
     # Test trustworthiness score.
     random_state = check_random_state(0)
 
     # Affine transformation
-    X = random_state.randn(100, 2)
+    X = random_state.randn(100, 2).astype(dtype)
     assert trustworthiness(X, 5.0 + X / 10.0) == 1.0
```

Comment:
unrelated to the purpose of this PR, but we should tolerate arithmetic rounding errors:

```suggestion
    assert trustworthiness(X, 5.0 + X / 10.0) == pytest.approx(1.0)
```